### PR TITLE
Fix tests that don’t clean up resources

### PR DESCRIPTION
**Pull request by robertjd**

### CODE REVIEW

File: test/it/helpers.js [hunk 1]
```diff
@@ -100,6 +100,27 @@ function cleanupOldApplications(callback) {
...
+ * Deletes an application and any account stores that are mapped to it
+ * @param  {Application} application A stormpath application object
+ * @param  {Function} callback
+ */
+function cleanupApplicationAndStores(application, callback) {
+  application.getAccountStoreMappings(function (err, mappings) {
+    if(err){
+      return callback(err);
+    }
+    mappings.each(function(mapping, next){
```
Comment (rdegges) [inline test/it/helpers.js]: space before {

File: test/it/helpers.js [hunk 2]
```diff
@@ -100,6 +100,27 @@ function cleanupOldApplications(callback) {
 }
 
 /**
+ * Deletes an application and any account stores that are mapped to it
+ * @param  {Application} application A stormpath application object
+ * @param  {Function} callback
+ */
+function cleanupApplicationAndStores(application, callback) {
+  application.getAccountStoreMappings(function (err, mappings) {
+    if(err){
```
Comment (rdegges) [inline test/it/helpers.js]: space before {

File: test/it/helpers.js [hunk 3]
```diff
@@ -100,6 +100,27 @@ function cleanupOldApplications(callback) {
...
+      return callback(err);
+    }
+    mappings.each(function(mapping, next){
+      mapping.getAccountStore(function(err, store){
```
Comment (rdegges) [inline test/it/helpers.js]: space before {

File: test/it/helpers.js [hunk 4]
```diff
@@ -100,6 +100,27 @@ function cleanupOldApplications(callback) {
...
+        if(err){
```
Comment (rdegges) [inline test/it/helpers.js]: space before {

File: test/it/helpers.js [hunk 5]
```diff
@@ -100,6 +100,27 @@ function cleanupOldApplications(callback) {
...
+          return next(err);
+        }
+        store.delete(next);
+      });
+    },callback);
```
Comment (rdegges) [inline test/it/helpers.js]: space after comma

File: test/it/jwt_authenticator_it.js [hunk 6]
```diff
@@ -104,7 +104,12 @@ describe('JwtAuthenticator',function(){
   });
 
   after(function(done){
-    application.delete(done);
+    helpers.cleanupApplicationAndStores(application,function(err){
```
Comment (rdegges) [inline test/it/jwt_authenticator_it.js]: space before {

File: test/it/oauth_authenticator_it.js [hunk 7]
```diff
@@ -102,7 +102,12 @@ describe('OAuthAuthenticator',function(){
   });
 
   after(function(done){
-    application.delete(done);
+    helpers.cleanupApplicationAndStores(application,function(err){
```
Comment (rdegges) [inline test/it/oauth_authenticator_it.js]: space before { and space after comma

File: test/it/saml_idp_url_builder_it.js [hunk 8]
```diff
@@ -26,6 +26,10 @@ describe('SamlIdpUrlBuilder', function () {
     });
   });
 
+  after(function(done){
```
Comment (rdegges) [inline test/it/saml_idp_url_builder_it.js]: space before {

File: test/it/tenant_it.js [hunk 9]
```diff
@@ -47,9 +47,8 @@ describe('Tenant', function() {
             directory = _directory;
             directory.createAccount(
               fakeAccount,
-              function(err,_account){
+              function(err){
```
Comment (rdegges) [inline test/it/tenant_it.js]: space before {

File: test/it/tenant_it.js [hunk 10]
```diff
@@ -64,6 +63,10 @@ describe('Tenant', function() {
       });
     });
 
+    after(function(done){
```
Comment (rdegges) [inline test/it/tenant_it.js]: space before {

File: test/it/tenant_it.js [hunk 11]
```diff
@@ -106,6 +110,10 @@ describe('Tenant', function() {
...
```
Comment (rdegges) [inline test/it/tenant_it.js]: space before {

File: test/it/client_credential_auth_it.js [hunk 12]
```diff
@@ -34,13 +34,7 @@ describe('Client Credential Authentication',function(){
...
 
   after(function(done){
-
-    account.delete(function(){
-      app.delete(function(){
-        done();
-      });
-    });
-
+    helpers.cleanupApplicationAndStores(app,done);
```
Comment (rdegges) [inline test/it/client_credential_auth_it.js]: space after comma

File: test/it/api_keys_it.js [hunk 13]
```diff
@@ -29,11 +29,7 @@ describe('Account api keys',function(){
   });
 
   after(function(done){
-    account.delete(function(){
-      app.delete(function(){
-        done();
-      });
-    });
+    helpers.cleanupApplicationAndStores(app,done);
```
Comment (rdegges) [inline test/it/api_keys_it.js]: space after comma

File: test/it/api_auth_it.js [hunk 14]
```diff
@@ -34,11 +34,7 @@ describe('Application.authenticateApiRequest',function(){
   });
 
   after(function(done){
-    account.delete(function(){
-      app.delete(function(){
-        done();
-      });
-    });
+    helpers.cleanupApplicationAndStores(app,done);
```
Comment (rdegges) [inline test/it/api_auth_it.js]: space after comma

**Comment by robertjd** [comment]
Thanks for the style review.  I'm going to pass on those for now.  We haven't yet added eslint to this repo, we can fix all the style issues when that is done.

**Comment by rdegges** [comment]
Left some small style comments per our usual. Other than those, +1!

**Comment by robertjd** [comment]
Eslint: #366

**Pull request merged**